compiler/src/dotty/tools/dotc/core/Types.scala

```diff
@@ -1749,7 +1749,7 @@ object Types {
         else NoType
       case t if defn.isNonRefinedFunction(t) =>
         t
-      case t if defn.isErasedFunctionType(t) =>
+      case t @ defn.PolyOrErasedFunctionOf(_) =>
         t
```

Comment:
Looks like this could be just ` case t if defn.isFunctionType(t) =>`

Reply:
This would make `findFunctionType` identify dependent function refinements. It looks like a desirable change.